glib/src/future_with_timeout.rs

```diff
@@ -0,0 +1,95 @@
+// Take a look at the license at the top of the repository in the LICENSE file.
+
+use futures_util::{
+    future::{self, Either, Future},
+    pin_mut,
+};
+
+// rustdoc-stripper-ignore-next
+/// The error returned when a future times out.
+pub struct FutureWithTimeoutError;
```

Comment:
Implement `std:error::Error` and `Display` on this please :)

Reply:
I added `Clone`, `Copy`, `PartialEq`, `Eq`, `Debug`, `Display`, `Error`;